mantle/cmd/ore/gcloud/deprecate-image.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcloud
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/coreos/mantle/platform/api/gcloud"
+	"github.com/spf13/cobra"
+)
+
+var (
+	cmdDeprecateImage = &cobra.Command{
+		Use:   "deprecate-image --image-name=ImageName [--state=DeprecationState]",
+		Short: "Deprecate GCP image",
+		Long:  "Change deprecation status of existing GCP image",
+		Run:   runDeprecateImage,
+	}
+
+	deprecateImageName        string
+	deprecateImageState       string
+	deprecateImageReplacement string
+)
+
+func init() {
+	cmdDeprecateImage.Flags().StringVar(&deprecateImageName,
+		"image-name", "", "GCP image name")
```

Comment:
btw I couldn't figure out either what the global gcloud image option is used for or how to grab it and use it in this file:

https://github.com/coreos/coreos-assembler/blob/b699ee1bfd5051fabe5b9fc3ae68f252f9ae83b4/mantle/cmd/ore/gcloud/gcloud.go#L42

This kept me from just using `--image` as the argument.

Reply:
You should just be able to reference `opts.Image` to get the value that was passed into the `--image` argument (which is on all commands of the `GCloud` variable.

Reply:
ok I'm not using `opts.Image`, but this did require me to slightly change the init code. See the `mantle/gcloud: don't mutate Image arg if not provided` commit.